images/cluster-setup/install.sh

```
@@ -122,33 +122,28 @@ install_tektoncd() {
   printf "\n"
 }
 
-wait_for_resource() {
-  args=( "$@" )
-  while ! kubectl get "${args[@]}" >/dev/null 2>/dev/null; do
-    sleep 10
-  done
-}
-
-wait_for_pod() {
-  args=( "$@" )
-  export -f wait_for_resource
-  if ! timeout 20s bash -c "wait_for_resource pods ${args[*]}"; then
-      exit_error "Pod not found."
-  fi
-  podname="$(kubectl get pods "${args[@]}" -o jsonpath='{.items[0].metadata.name}')"
-  kubectl wait --for=condition=Ready "pod/$podname" -n openshift-pipelines --timeout=60s >/dev/null
-}
-
 postchecks() {
   printf "Checking deployment\n"
   #checking if the pipelines and triggers pods are up and running
   for i in "${!clusters[@]}"; do
     switch_cluster
     printf "  - %s:\n" "${clusters[$i]}"
-    for pod in tekton-pipelines-controller tekton-triggers-controller tekton-triggers-core-interceptors; do
-      printf "    - %s: " "$pod"
-      wait_for_pod -n openshift-pipelines -l=app="$pod"
-      printf "OK\n"
+    for deploy in tekton-pipelines-controller tekton-triggers-controller tekton-triggers-core-interceptors; do
```

If any (or all) those deployment failed, I do not see an error code being returned when the script exits. That could trigger silent failures.

---

Fixed.